GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -141,6 +141,31 @@ string ComposeOutput()
                 () => executable.RunCommandAsync(arguments, input, createWindow));
         }
 
+        /// <summary>
+        /// Launches a process for the executable and returns <c>true</c> if its exit code is zero. Fix #6593.
+        /// </summary>
+        /// <remarks>
+        /// This method uses <see cref="RunCommand"/> to execute multiple commands in batch, used in accordance with
+        /// <see cref="ArgumentBuilderExtensions.BuildBatchArguments(ArgumentBuilder, IEnumerable{string}, int, int)"/>
+        /// to work around Windows command line length 32767 character limitation
+        /// <see href="https://docs.microsoft.com/en-us/windows/desktop/api/processthreadsapi/nf-processthreadsapi-createprocessa"/>.
+        /// </remarks>
+        /// <param name="executable">The executable from which to launch a process.</param>
+        /// <param name="batchArguments">The array of batch arguments to pass to the executable</param>
+        /// <param name="input">Bytes to be written to the process's standard input stream, or <c>null</c> if no input is required.</param>
+        /// <param name="createWindow">A flag indicating whether a console window should be created and bound to the process.</param>
+        /// <returns><c>true</c> if the process's exit code was zero, otherwise <c>false</c>.</returns>
+        [MustUseReturnValue("Callers should verify that " + nameof(RunBatchCommand) + " returned true")]
+        public static bool RunBatchCommand(
+            this IExecutable executable,
+            ArgumentString[] batchArguments,
+            byte[] input = null,
+            bool createWindow = false)
+        {
+            // TODO: Process somehow always return non-zero exit code.
```

Comment:
I'm afraid, the limit is lower. The Win32 API doc says: If the executable is left empty and added to the arguements, the length limit was `MAXPATH`.
`[TestCase(8193 - 1 - 19, 1, 10)]` is the maximum that succeeds.

Reply:
`MAXPATH` length limit in Win32 API doc is for full path of the `git.exe` executable module, whereas `lpCommandLine` max length is always `32,768 characters, including the Unicode terminating null character`. Real test leaves us at `short.MaxValue - 1 // 32766` though.
>
> `lpCommandLine`
> The maximum length of this string is 32,768 characters, including the Unicode terminating null character. If `lpApplicationName` is NULL, the module name portion of `lpCommandLine` is limited to MAX_PATH characters.
>

The TODO was left here because in the test case `cmd.exe /C echo` always return non-zero exit code. In debug, `git.exe` returns correctly, so this is for the test case only. I will investigate the test case and resolve this.

Reply:
Yes, you're right. It seems to be a limitation of the length of `<command>` in `cmd /c <command>`.

Reply:
You're right, I need to find another executable to test, `cmd.exe` command line length limitation is `8191` characters.
https://support.microsoft.com/en-us/help/830473/command-prompt-cmd-exe-command-line-string-limitation

Reply:
I pass max length `8191` into the test case and everything works find now. About order execution of and branch, I've switch function execution before boolean var, so in any language (C/C++), it would look logical.